src/session/SessionStore.ts

```diff
@@ -1,109 +1,66 @@
-"use strict";
-
-import ChLogger from "ch-logger";
-import moment from "moment";
-
-import Encoding from "../encoding";
-import cache from "../cache";
-import config from "../config";
-import ApplicationLogger from "ch-logger/lib/ApplicationLogger";
-import Session from "./Session";
-import SessionKeys from "./SessionKeys";
-import AccessTokenData from "./AccessTokenData";
-
-class SessionStore {
-
-    private _idOctets = 21;
-    private _sessionExpiredError = "Session has expired.";
-
-    private _logger: ApplicationLogger;
-
-    constructor(logger: ApplicationLogger) {
-        this._logger = logger;
+import { Cache } from "../cache/Cache";
+import { SessionKeys } from "./SessionKeys";
+import { AccessToken } from "./model/AccessToken";
+import { Encoding } from "../encoding/Encoding";
+import { VerifiedSession, Session } from "./model/Session";
+import { ISignInInfo } from "./model/ISignInInfo";
+import { EitherAsync, Either } from "purify-ts";
+import { Failure } from "../error/FailureType";
+import {
+    liftToAsyncEither,
+    liftEitherToAsyncEither,
+    liftFunctionToAsyncEither,
+    liftEitherFunctionToAsyncEither,
+} from "../utils/EitherUtils";
+import { Cookie } from "./model/Cookie";
+
+export class SessionStore {
+
+
+    public constructor(private readonly cache: Cache) {
     }
 
-    private getAccessTokenData(data: any) {
+    private getAccessTokenData(session: Session): AccessToken | undefined {
 
-        const rawAccessTokenData = data[SessionKeys.AccessToken];
+        const rawAccessTokenData: ISignInInfo | undefined = session.data[SessionKeys.SignInInfo];
 
-        return rawAccessTokenData ?
-            new AccessTokenData(rawAccessTokenData) :
-            undefined;
+        return rawAccessTokenData ? rawAccessTokenData[SessionKeys.AccessToken] : undefined;
     }
 
-    async load(sessionId: string): Promise<Session> {
-
-        let encodedData;
-
-        try {
-            encodedData = await cache.get(sessionId);
-
-        } catch (error) {
-
-            this._logger.error(error.message);
-            throw new Error("Error trying to retrieve from cache");
-        }
-
-        const data = await this.decodeSession(encodedData);
-
-        const accessTokenData = this.getAccessTokenData(data);
-
-        if (accessTokenData && accessTokenData.expiresIn) {
-            this.validateExpiration(accessTokenData.expiresIn);
-        }
-
-        return new Session(sessionId, accessTokenData);
-    }
+    public loadSession = (sessionSecret: string, cookie: string): EitherAsync<Failure, VerifiedSession> => {
```

Comment:
I feel that function that loads session should not be responsible for cookie verification. It is one more responsibility too many for the method.